src/main/java/reactor/core/publisher/FluxMapOrFilterFuseable.java

```diff
@@ -0,0 +1,450 @@
+			int m;
+			if ((requestedMode & Fuseable.THREAD_BARRIER) != 0) {
+				if ((requestedMode & Fuseable.SYNC) != 0) {
+					m = s.requestFusion(Fuseable.SYNC);
+				} else {
+					m = Fuseable.NONE;
+				}
+			} else {
+				m = s.requestFusion(requestedMode);
+			}
+			sourceMode = m;
+			return m;
+		}
+
+		@Override
+		public int size() {
+			return s.size();
+		}
+	}
+
+	static final class MapOrFilterFuseableConditionalSubscriber<T, R>
+			implements ConditionalSubscriber<T>, Receiver, Producer, Loopback,
+			           SynchronousSubscription<R>, Trackable {
+		final ConditionalSubscriber<? super R>			actual;
+		final Function<? super T, ? extends R> mapper;
+
+		boolean done;
+
+		QueueSubscription<T> s;
+
+		int sourceMode;
+
+		public MapOrFilterFuseableConditionalSubscriber(ConditionalSubscriber<? super R> actual, Function<? super T, ? extends R> mapper) {
+			this.actual = actual;
+			this.mapper = mapper;
+		}
+
+		@SuppressWarnings("unchecked")
+		@Override
+		public void onSubscribe(Subscription s) {
+			if (Operators.validate(this.s, s)) {
+				this.s = (QueueSubscription<T>)s;
+				actual.onSubscribe(this);
+			}
+		}
+
+		@Override
+		public void onNext(T t) {
+			if (done) {
+				Exceptions.onNextDropped(t);
+				return;
+			}
+
+			int m = sourceMode;
+			
+			if (m == 0) {
+				R v;
+	
+				try {
+					v = mapper.apply(t);
+				} catch (Throwable e) {
+					onError(Exceptions.mapOperatorError(s, e, t));
+					return;
+				}
+	
+				if (v == null) {
+					s.request(1);
+				} else {
+					actual.onNext(v);
+				}
+
+			} else
+			if (m == 2) {
+				actual.onNext(null);
+			}
+		}
+
+		@Override
+		public boolean tryOnNext(T t) {
+			if (done) {
+				Exceptions.onNextDropped(t);
+				return true;
+			}
+
+			int m = sourceMode;
+			
+			if (m == 0) {
+				R v;
+	
+				try {
+					v = mapper.apply(t);
+				} catch (Throwable e) {
+					onError(Exceptions.mapOperatorError(s, e, t));
+					return true;
+				}
+	
+				if (v == null) {
+					return false;
+				}
+	
+				return actual.tryOnNext(v);
+			} else
+			if (m == 2) {
+				actual.onNext(null);
+			}
+			return true;
+		}
+
+		
+		@Override
+		public void onError(Throwable t) {
+			if (done) {
+				Exceptions.onErrorDropped(t);
+				return;
+			}
+
+			done = true;
+
+			actual.onError(t);
+		}
+
+		@Override
+		public void onComplete() {
+			if (done) {
+				return;
+			}
+			done = true;
+
+			actual.onComplete();
+		}
+
+		@Override
+		public boolean isStarted() {
+			return s != null && !done;
+		}
+
+		@Override
+		public boolean isTerminated() {
+			return done;
+		}
+
+		@Override
+		public Object downstream() {
+			return actual;
+		}
+
+		@Override
+		public Object connectedInput() {
+			return mapper;
+		}
+
+		@Override
+		public Object upstream() {
+			return s;
+		}
+		
+		@Override
+		public void request(long n) {
+			s.request(n);
+		}
+		
+		@Override
+		public void cancel() {
+			s.cancel();
+		}
+
+		@Override
+		public R poll() {
+			T v = s.poll();
```

Comment:
reactor.core.publisher.FluxMapFuseable.MapFuseableSubscriber#poll doesn't do any of that, I assume thats because there is no conditional request() etc though? Or is this missing from FluxMapFuseable?